### PR TITLE
Fix conditional hooks

### PR DESCRIPTION
**Pull request by rodrigoncalves**
Fixes issue described on #266.

### CODE REVIEW

File: src/app/proposals/hooks/useGetProposalDeadline.ts [hunk 1]
```diff
@@ -1,34 +1,61 @@
...
+    }
+  }, [fetchOnMount])
+  /**
+   * Hook to fetch latest block number if passedLatestBlockNumber is undefined
+   */
+  useEffect(() => {
+    if (!passedLatestBlockNumber && fetchOnMount) {
+      fetchLatestBlockNumber()
+    }
+  }, [passedLatestBlockNumber])
```
Comment (rodrigoncalves) [inline src/app/proposals/hooks/useGetProposalDeadline.ts:44]: This gives warning `react-hooks/exhaustive-deps`

File: src/app/proposals/hooks/useGetProposalDeadline.ts [hunk 2]
```diff
@@ -1,34 +1,61 @@
...
+   * When the fetchOnMount changes, fetch deadline block and latest block number
+   */
+  useEffect(() => {
+    if (fetchOnMount) {
+      fetchProposalDeadlineBlock()
+      if (!passedLatestBlockNumber) {
+        fetchLatestBlockNumber()
+      }
+    }
+  }, [fetchOnMount])
```
Comment (rodrigoncalves) [inline src/app/proposals/hooks/useGetProposalDeadline.ts:36]: This gives warning `react-hooks/exhaustive-deps`

**Review by rodrigoncalves — COMMENTED** [review]
I love the approach. Just a few minor changes.

**Comment by rodrigoncalves** [comment]
Reopening this, I don't why it was closed.

**Review by LordCheta — APPROVED** [review]
LGTM

**Pull request merged**